### PR TITLE
Supporting img component to load data from context asset_source resources

### PR DESCRIPTION
**Pull request by ihavecoke**
Release Notes:

- Fixed gpui img component does not support relative paths

I discovered a bug in the gpui img component where it only supports absolute paths, not relative paths. For example, in the screenshots below, running the images.rs file in the gpui examples folder shows that the app-icon.png on the left failed to load, while the image on the right loaded successfully via URL.

image

This issue because the gpui img component uses the fs::read method to load images with PathBuf type parameters, which fails to resolve relative paths:

 ```rust
fs::read((PathBuf::from_str("examples/image/app-icon.png").unwrap()))?
 ```
 
 The code above results in an error as the file is not found:
 
```shell
 [2024-07-27T01:20:03Z ERROR gpui] Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })
```

However, when using an absolute path the image loads successfully:

```rust
fs::read((PathBuf::from_str("/Users/xxx/xxx/xxx/gpui/examples/image/app-icon.png").unwrap()))?
```

Here is a screenshot of the code execution:

image

To fix this issue, the img component could either convert relative paths to absolute paths before loading, or directly support relative paths. I am considering submitting a PR to address this.

Another potential solution is to use the context’s global access_source method to load images. This allows for loading images with relative paths, similar to how icon resources are loaded. Below is the modified code and the result when running in the avatar storybook. The img component supports PathBuf parameters by first checking the zed assets directory for the resource before falling back to the existing logic:

image

In summary, the gpui img component currently does not support relative paths with PathBuf parameters and requires absolute paths. I have added functionality to load images using relative paths by checking the context‘s access_source first, integrating it with the project’s assets loading mechanism.

### CODE REVIEW

**Comment by maxdeviant** [comment]
This PR looks like it's just a reskinning of https://github.com/zed-industries/zed/pull/15292 with a different explanation.

I still don't agree that this is the right solution.

The local image example does work when run from within the `crates/gpui` directory:

```sh
cd crates/gpui
cargo run --example image
```

Screenshot 2024-07-26 at 9 47 24 PM

If this is an issue you are facing, please open a GitHub issue to discuss it.

**Comment by ihavecoke** [comment]
@maxdeviant Thank you for your response.
Indeed, when I previously ran image.rs directly in RustRover IDE, the icon on the left did not render. However, using the method you provided:

```shell
cargo run --example images
```
the icon rendered correctly. 

I initially thought this was a bug.

**Comment by ihavecoke** [comment]
@maxdeviant I started a discussion regarding an [issue](https://github.com/zed-industries/zed/issues/15335) with using the img component, and I hope to get some help from the community. Thank you very much!
